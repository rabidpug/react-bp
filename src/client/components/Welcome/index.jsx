import React, { Component, } from 'react';

import HelloButton from '../HelloButton';
import { connect, } from 'react-redux';
import gStyles from 'Styles/global';
import { hot, } from 'react-hot-loader';
import mapWelcome from './map';

@hot( module )
@connect( mapWelcome.State )
export default class Welcome extends Component {
  render () {
    const { style, message, } = this.props;

    return (
      <div
        className={ gStyles.cardStyle }
        style={ { ...style, } }>
        <div className={ gStyles.cardHeader }>
          <h1>Welcome Message</h1>
        </div>
        <div className={ gStyles.cardTop }>
          <p>{message}</p>
        </div>
        <div className={ gStyles.cardBottom }>
          <HelloButton />
        </div>
      </div>
    );
  }
}
