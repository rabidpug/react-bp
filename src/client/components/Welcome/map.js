import { getGreeterMessage, } from 'Store/greeter/selectors';

const mapWelcome = { State: state => ( { message: getGreeterMessage( state ), } ), };

export default mapWelcome;
