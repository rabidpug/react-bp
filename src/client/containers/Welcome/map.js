import store from 'Store';
const map = {
  State: state => ( {
    messages : store.greeter.get.messages( state ),
    publicId : store.user.get.publicId( state ),
  } ),
};

export default map;
