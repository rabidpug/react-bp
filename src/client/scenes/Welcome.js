import FadeWrap from '../animations/FadeWrap';
import HelloButton from 'Containers/HelloButton';
import HelloMessage from 'Containers/HelloMessage';
import React from 'react';

const Welcome = FadeWrap( ( { style, } ) => (
  <div style={ { ...style, } }>
    <HelloMessage />
    <HelloButton />
  </div>
) );

export default Welcome;
