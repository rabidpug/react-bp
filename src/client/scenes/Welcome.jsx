import HelloMessage from 'Components/Message/HelloMessage';

const Welcome = HelloMessage;

export default Welcome;
