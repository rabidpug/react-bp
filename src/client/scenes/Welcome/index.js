import HelloButton from '../../containers/HelloButton';
import Message from '../../containers/Message';
import React from 'react';

const Welcome = () => (
  <div>
    <Message />
    <HelloButton />
  </div>
);

export default Welcome;
