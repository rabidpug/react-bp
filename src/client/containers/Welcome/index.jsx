import React, { Component, } from 'react';

import Card from 'Components/Card';
import Message from 'Components/Message';
import SendMessage from 'Containers/SendMessage';
import { connect, } from 'react-redux';
import { hot, } from 'react-hot-loader';
import map from './map';

@hot( module )
@connect( map.State )
export default class Welcome extends Component {
  componentDidMount () {
    this.scrollToBottom();
  }

  componentDidUpdate () {
    this.scrollToBottom();
  }

  scrollToBottom = () => {
    this.messagesEnd.scrollIntoView( { behavior: 'smooth', } );
  };

  render () {
    const { messages, publicId, } = this.props;

    return (
      <Card>
        <Card.Header>
          <h1 style={ { margin: 0, } }>Welcome Messaging</h1>
        </Card.Header>
        <Card.Body>
          {messages.length === 0
            ? <p style={ { width: '100%', } }>No Messages</p>
            :             messages.map( value => (
              <Message
                isUser={ value.userProfile.publicId === publicId }
                { ...value }
                key={ value.timestamp + value.userProfile.displayNames }
              />
            ) )
          }

          <div
            ref={ el => {
              this.messagesEnd = el;
            } }
            style={ {
              clear : 'both',
              float : 'left',
            } }
          />
        </Card.Body>
        <Card.Footer>
          <SendMessage />
        </Card.Footer>
      </Card>
    );
  }
}
