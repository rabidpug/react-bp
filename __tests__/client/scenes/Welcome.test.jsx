import React from 'react';
import Welcome from 'Scenes/Welcome';
import configureStore from 'redux-mock-store';
import { shallow, } from 'enzyme';
import { shallowToJson, } from 'enzyme-to-json';

const mockStore = configureStore();

describe(
  'TopBar', () => {
    const store = mockStore( {} );

    store.dispatch = jest.fn();

    it(
      'Should render the Welcome page', () => {
        const wrapper = shallow( <Welcome store={ store } /> );

        expect( shallowToJson( wrapper ) ).toMatchSnapshot();
      }
    );
  }
);
