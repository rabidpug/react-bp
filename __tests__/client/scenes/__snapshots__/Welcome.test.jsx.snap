// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar Should render the Welcome page 1`] = `
<Message
  dispatch={[MockFunction]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [MockFunction],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
